infrastructure/src/main/java/org/corfudb/infrastructure/ServerContext.java

```diff
@@ -440,10 +442,30 @@ public long getStartingAddress() {
         return startingAddress == null ? 0 : startingAddress;
     }
 
+
     public void setStartingAddress(long startingAddress) {
         dataStore.put(Long.class, PREFIX_STARTING_ADDRESS, KEY_STARTING_ADDRESS, startingAddress);
     }
 
+    /**
+     * Fetch last globally committed address for this server context.
+     *
+     * @return global address
+     */
+    public long getLastCommittedAddress() {
+        Long address = dataStore.get(Long.class, PREFIX_LOGUNIT, ADDRESS_JOURNAL);
+        return address == null ? Address.NON_ADDRESS : address;
+    }
+
+    /**
+     * Set the last globally committed address for this server context.
+     *
+     * @param address the address to set
+     */
+    public synchronized void setLastCommittedAddress(long address) {
```

Comment:
The file based datastore was not built for high throughput. At least I never put in the effort to do make it efficient. I am worried that this could have some unintended affects on performance if the datastore is not performant enough.